app/helpers/application_helper.rb

```diff
@@ -406,4 +406,35 @@ def preloaded_json
     return '{}' if @preloaded.blank?
     @preloaded.transform_values { |value| escape_unicode(value) }.to_json
   end
+
+  def client_side_setup_data
+    service_worker_url = Rails.env.development? ? 'service-worker.js' : Rails.application.assets_manifest.assets['service-worker.js']
+    current_hostname_without_port = RailsMultisite::ConnectionManagement.current_hostname.sub(/:[\d]*$/, '')
+
+    setup_data = {
+      cdn: Rails.configuration.action_controller.asset_host,
+      base_url: current_hostname_without_port,
+      base_uri: Discourse::base_uri,
+      environment: Rails.env,
+      letter_avatar_version: LetterAvatar.version,
+      markdown_it_url: asset_url('markdown-it-bundle.js'),
+      service_worker_url: service_worker_url,
+      default_locale: SiteSetting.default_locale,
+      asset_version: Discourse.assets_digest,
+      disable_custom_css: loading_admin?,
+      highlight_js_path: HighlightJs.path,
+      enable_s3_uploads: SiteSetting.Upload.enable_s3_uploads
+    }
+
+    if guardian.can_enable_safe_mode? && params["safe_mode"]
+      setup_data[:safe_mode] = normalized_safe_mode
+    end
+
+    if SiteSetting.Upload.enable_s3_uploads
+      setup_data[:s3_cdn] = SiteSetting.Upload.s3_cdn_url.presence
```

Comment:
Hmm we used to check if `SiteSetting.Upload.s3_cdn_url.present?`

Reply:
Yea, took the liberty to use `.presence` because we run an empty check on the value anyway. 😅 but I can change this back

https://github.com/discourse/discourse/blob/1a64b3a4871cceb1b42d5c54ada58ab1241022f6/app/assets/javascripts/discourse.js.es6#L36

Reply:
Ahh icic then don't worry about this comment. :+1: